pulp_ansible/tests/unit/test_search.py

```diff
@@ -0,0 +1,157 @@
+            for j in range(0, 3):
+                collection_name = "".join(
+                    [random.choice(string.ascii_lowercase) for x in range(0, 5)]
+                )
+                for v in range(1, 3):
+                    vstring = f"1.0.{v}"
+                    tarfn = make_cv_tarball(namespace_name, collection_name, vstring)
+                    self.collections[(namespace_name, collection_name, vstring)] = {
+                        "tags": ["tag" + namespace_name + collection_name + vstring],
+                        "tar": tarfn,
+                    }
+
+        # "import" the collections
+        specs = sorted(list(self.collections.keys()))
+        for ids, spec in enumerate(specs):
+            cdata = self.collections[spec]
+
+            this_repo_name = random.choice(list(self.repos.keys()))
+            this_repo = self.repos[this_repo_name]
+            self.collections[spec]["repo"] = this_repo
+            self.collections[spec]["repo_name"] = this_repo_name
+
+            # make an artifact
+            rawbin = open(cdata["tar"], "rb").read()
+            artifact = Artifact.objects.create(
+                sha224=hashlib.sha224(rawbin).hexdigest(),
+                sha256=hashlib.sha256(rawbin).hexdigest(),
+                sha384=hashlib.sha384(rawbin).hexdigest(),
+                sha512=hashlib.sha512(rawbin).hexdigest(),
+                size=os.path.getsize(tarfn),
+                file=SimpleUploadedFile(tarfn, rawbin),
+            )
+            artifact.save()
+
+            # make the collection
+            col, _ = Collection.objects.get_or_create(name=spec[0])
+            col.save()
+            self.collections[spec]["col"] = col
+
+            # make the collection version
+            cv = CollectionVersion(collection=col, namespace=spec[0], name=spec[1], version=spec[2])
+            cv.save()
+            self.collections[spec]["cv"] = cv
+
+            # add tags ...
+            #   ansible_collectionversion -> ansible_collectionversion_tags -> ansible_tags
+            for tag_name in cdata["tags"]:
+                this_tag, _ = Tag.objects.get_or_create(name=tag_name)
+                cv.tags.add(this_tag)
+                cv.save()
+
+            # trigger an update and rebuild of the search vector
+            #   THIS IS THE ONLY WAY THAT THE SEARCH VECTOR IS CREATED!
+            cv.is_highest = False
+            cv.save()
+
+            # bind the artifact to the cv
+            ca = ContentArtifact.objects.create(
+                artifact=artifact, content=cv, relative_path=cv.relative_path
+            )
+            ca.save()
+            self.collections[spec]["ca"] = ca
+
+            # add the cvs to a repository version
+            qs = CollectionVersion.objects.filter(pk=cv.pk)
+            with this_repo.new_version() as new_version:
+                new_version.add_content(qs)
```

Comment:
Same with creating the ContentArtifact and adding the CV to the repo, I think these can all be removed as they are unrelated to the test.

Reply:
Yep, I'll try taking this out.